api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -93,6 +93,10 @@ public void setEncounterDAO(EncounterDAO dao) {
 	@Override
 	public Encounter saveEncounter(Encounter encounter) throws APIException {
 		
+		if(dao.getEncounters(new EncounterSearchCriteriaBuilder().setPatient(encounter.getPatient()).setFromDate(encounter.getDateCreated()).createEncounterSearchCriteria()).size() > 0) {
```

Comment:
Shouldn't you be calling encounter.getEncounterDatetime() instead of getDateCreated() in setFromDate call ?